libenv/sysinfo.c

```diff
@@ -785,8 +785,12 @@ static void GetNameInfo3(EvalContext *ctx)
         Log(LOG_LEVEL_ERR, "I don't understand what architecture this is");
     }
 
-    strcpy(workbuf, "compiled_on_");
-    strcat(workbuf, CanonifyName(AUTOCONF_SYSNAME));
+    char compile_str[] = "compiled_on_";
```

Comment:
Why not just `char *`? We are not using `char str[]` anywhere else, AFAICT.

Reply:
Because I wanted to use `sizeof`
Even though `char *` and `char[]` is the same when initialising on the stack with a string literal, `sizeof(char *)` will give you the size of a pointer, and `sizeof(char[])` will give you the size of the array.
It doesn't really matter much, but I think it's unnecessary to use `strlen` to figure out the length, and I would like to do it dynamically (as in not using some constant for the length).

Reply:
Fair enough.

Reply:
Interesting use of the term _dynamically_ ;)